src/jgikbase/idmapping/userlookup/kbase_user_lookup.py

```diff
@@ -91,3 +92,21 @@ def is_valid_user(self, username: Username) -> Tuple[bool, Optional[int], Option
         self._check_error(r)
         j = r.json()
         return (len(j) == 1, None, 3600)
+
+
+def build_lookup(config: Dict[str, str]) -> UserLookup:
```

Comment:
To me, the work this function is really doing is validating the configuration for building a KBaseUserLookup object, so maybe it could be named as such with the constructor separated out. I'm not adamant on this.

Reply:
I'm not quite following you here - what do you mean by having the constructor separated out?

The purpose of this function is providing a factory method so I can do:

```
m = importlib.import('modulename')
userlookup = m.build_lookup(config)
```

Reply:
I meant that it might be a bit more descriptive to do:
```py
validate_kbase_user_lookup_config(config)
userlookup = KBaseUserLookup(config['url'], Token(config['token']), config['admin-role'])
```
Not worth dwelling on though

Reply:
yeah, that doesn't work for the import behavior I mentioned above IIUC